### ISSUE TITLE
Schema types break under certain conditions

### ISSUE THREAD

**Issue by danielroe** [vueuse/head#138]
this may be a https://github.com/harlan-zw/zhead issue...

Here are some things that should be typed properly:

```js
useHead({
  link: [{ rel: 'mask-icon', href: '/safari-pinned-tab.svg', color: '#1a202c' }] // color is not allowed
})
```

```js
useHead({
  link: [
    ...[].map(() => ({ rel: 'prefetch', href: '' })), // this damages the type inference
    { rel: 'icon', type: 'image/svg', href: '/favicon.svg' },
  ]
})
```

**Comment by harlan-zw** [comment]
Thanks for the issue! The `color` was [fixed](https://github.com/harlan-zw/zhead/pull/5) in zhead yesterday, which is now in the latest RC (1.0.0-rc.8).

Not too sure what the issue is for number two. I've tried [replicating](https://github.com/vueuse/head/commit/ee49fcbf1c4a72494daed2a2c901666cb19912e6) it and the types seem to be coming through correctly. Is this via Nuxt or this package's `useHead`? 

![image](https://user-images.githubusercontent.com/5326365/195743760-573727e6-ae25-4ea7-9b75-0a5bfd4cd91d.png)

If you can provide some more details I'll try to get to the bottom of it

**Comment by danielroe** [comment]
Oh, interesting. This may be an issue with the Nuxt types in that case.

**Comment by harlan-zw** [comment]
I see, I'll take a look in Nuxt

**Comment by harlan-zw** [comment]
```
npx nuxi init nuxt-app
```

`yarn add -D nuxt@npm:nuxt3@latest`

PHPStorm - Works :+1: 

![image](https://user-images.githubusercontent.com/5326365/195815602-f25e6b3d-13ec-411e-b09b-c47c41c0c5e1.png)

Code - Does not work :-1: 

![image](https://user-images.githubusercontent.com/5326365/195822617-ab1bad3f-e71b-4cb5-944b-3183f00783df.png)

**Comment by harlan-zw** [comment]
Hey @danielroe 

Quick update, I was able to replicate the issue, but only through hacking the types. 

Since I can't replicate it reliably though, it's a bit hard to fix.

If you have any spare time, maybe you could do a quick sanity check of the checks. You can find them here: https://github.com/harlan-zw/zhead/blob/main/packages/schema-vue/src/types.ts

An example of how it's used:

```ts
interface Link {
  rel: string
  // ... etc

}  
type Links = MaybeDeeplyComputedRefArray<Link[]>
``

Possibly something is going wrong with the array inference? I'll think about how I can rewrite the types to be simpler which may fix it.

While I was testing, I noticed the auto-completion wasn't working for VSCode, which I've pushed a [fix](https://github.com/harlan-zw/zhead/commit/50a85cb75bf4b207cd3d2271fafc157317a2e545) for.

That's available in v1.0.0-rc.8, probably should make sure Nuxt uses that for the RC release if we're going to have the types as part of the release notes.

**Comment by harlan-zw** [comment]
The way that the Schema is typed is much simpler now which should resolve the issue.

Going to close off for now, if you spot the issue again please let me know :)
